### PR TITLE
Added reference for server.py file

### LINKED ISSUE #9363: Reference Guide should link to Users Guide where appropriate.
E.g. the `Slope` reference guide entry could link to the "Adding Anotations" User Guide section for a more complete discussion.

### PR DESCRIPTION
**Pull request by Med16-11**
- [x] issues: addresses #9363

### CODE REVIEW

File: src/bokeh/server/server.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
     :class:`~bokeh.application.application.Application` instances, and will
     automatically create and coordinate the lower level Tornado components.
 
+See :ref:`userguide_server_introduction` for information on server.
```
Comment (bryevdv) [inline src/bokeh/server/server.py]: This is a great place for a link, just a small suggested edit:

```suggestion
See :ref:`userguide_server_introduction` for general information on the Bokeh server.
```

**Comment by Med16-11** [comment]
Hey @bryevdv ! I need a little help. Even after creating pull request my repo shows this option of creating pull request and my commit doesn't get merge properly. Can you help me in resolving this issue? I have tried resolving it using this 
https://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch
but it doesn't work in my case.
image

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks for the PR @Med16-11 ! I've update the OP to remove "fixes" to avoid auto-closing the issue (it's a very general, ongoing issue to make changes in many places)

**Comment by tcmetzger** [comment]
@bryevdv @Med16-11 If I'm not mistaken, the correct reference would be `ug_server_introduction`, not `userguide_server_introduction`. The `userguide` reference probably came from when this PR was based on an older branch.
@Med16-11 would you mind doing a follow-up PR?

**Comment by Med16-11** [comment]
sure will do that @tcmetzger ! Also I wanted to work on few more documentation issues since I am applying for "Improve Bokeh's documentation using the Diátaxis framework", it would be very helpful if you can suggest some issues to work on.

**Pull request merged**